### PR TITLE
Add tests for integrare Results

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**
Relevant tests are in `src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx`.
The rest are nit changes.

### CODE REVIEW

File: src/__tests__/hooks/useFetchCompareResults.ts
Comment (beatrice-acasandrei) [inline src/__tests__/hooks/useFetchCompareResults.ts:1]: I believe this is a duplicate of `src/__tests__/hooks/useFetchCompareResults.test.ts`, it seems to be a mistaken leftover from rebasing.

File: src/common/constants.ts [hunk 1]
```diff
@@ -10,8 +10,6 @@ import {
 export const treeherderBaseURL = 'https://treeherder.mozilla.org';
 
 export const maxRevisionsError = 'Maximum 1 revision(s).';
-export const featureNotSupportedError =
```
Comment (beatrice-acasandrei) [inline src/common/constants.ts:13]: `featureNotSupportedError` is not used.

**Review by esanuandra — APPROVED** [review]

File: src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        title={Strings.metaData.pageTitle.results}
+      />,
+    );
+
+    expect(spyOnFetch).toHaveBeenCalledWith(
+      'https://treeherder.mozilla.org/api/perfcompare/results/?base_repository=mozilla-central&base_revision=6089e7f0fa57a29c6d080f135f65e146c34457d8&new_repository=mozilla-central&new_revision=1d5eb1343cc87a9be3dfe4b884822506ffdda7d3&framework=1&interval=86400&no_subtests=true',
+    );
+  });
+
+  it('Should reject fetchCompareResults if fetch returns no results', () => {
```
Comment (julienw) [inline src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx:46]: The title isn't clear to me, I don't see this in the test body.

**Review by julienw — COMMENTED** [review]
I believe we need some expectation about the HTML rendering. It could be as simple as `expect(document.body).toMatchSnapshot()` for example.

File: src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  it('Should fetch and display recent results', () => {
+    const { testCompareData } = getTestData();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => ({
+          results: testCompareData,
+        }),
+      }),
+    ) as jest.Mock;
+    const spyOnFetch = jest.spyOn(global, 'fetch');
```
[the remark this answers is not in the document]
Comment (beatrice-acasandrei) [reply src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx]: There are a few tests where this is happening so I removed the constant from those where it was straight forward to do so, hope that's fine.

**Review by esanuandra — APPROVED** [review]

File: src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx [hunk 4]
```diff
@@ -61,9 +65,10 @@ describe('Results View/fetchCompareResults', () => {
       />,
     );
 
-    expect(spyOnFetch).toHaveBeenCalledWith(
+    expect(global.fetch).toHaveBeenCalledWith(
       'https://treeherder.mozilla.org/api/perfcompare/results/?base_repository=mozilla-central&base_revision=6089e7f0fa57a29c6d080f135f65e146c34457d8&new_repository=mozilla-central&new_revision=1d5eb1343cc87a9be3dfe4b884822506ffdda7d3&framework=1&interval=86400&no_subtests=true',
     );
     expect(store.getState().compareResults.data).toStrictEqual({});
+    expect(document.body).toMatchSnapshot();
```
Comment (julienw) [inline src/__tests__/CompareResults/beta/fetchCompareResults.test.tsx:72]: (optional) Maybe the snapshot would be easier to read with somethig such as:

```js
expect(await screen.findByTestId("results-table")).toMatchSnapshot();
```
but that's up to you!

**Review by julienw — APPROVED** [review]

**Pull request merged**